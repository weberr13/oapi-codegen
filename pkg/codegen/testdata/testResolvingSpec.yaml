openapi: 3.0.1

info:
  title: OpenAPI-CodeGen Test
  description: 'This is a test OpenAPI Spec'
  version: 1.0.0

servers:
- url: https://test.oapi-codegen.com/v2
- url: http://test.oapi-codegen.com/v2

paths:
  /test/{name}:
    get:
      tags:
      - test
      summary: Get test
      operationId: getTestByName
      parameters:
      - name: name
        in: path
        required: true
        schema:
          type: string
      responses:
        200:
          description: Success
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Test'

components:
  schemas:
    Test:
      properties:
        name:
          type: string
        verifier:
          $ref: 'http://localhost:23456/included.yaml#/components/schemas/Verifier'
        cases:
          type: array
          items:
            $ref: '#/components/schemas/TestCase'

    TestCase:
      properties:
        name:
          type: string
        command:
          type: string

