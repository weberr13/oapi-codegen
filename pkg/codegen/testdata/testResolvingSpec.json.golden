{
  "components": {
    "schemas": {
      "Test": {
        "properties": {
          "cases": {
            "items": {
              "properties": {
                "command": {
                  "type": "string"
                },
                "name": {
                  "type": "string"
                }
              }
            },
            "type": "array"
          },
          "name": {
            "type": "string"
          },
          "verifier": {
            "properties": {
              "name": {
                "type": "string"
              }
            }
          }
        }
      },
      "TestCase": {
        "properties": {
          "command": {
            "type": "string"
          },
          "name": {
            "type": "string"
          }
        }
      }
    }
  },
  "info": {
    "description": "This is a test OpenAPI Spec",
    "title": "OpenAPI-CodeGen Test",
    "version": "1.0.0"
  },
  "openapi": "3.0.1",
  "paths": {
    "/test/{name}": {
      "get": {
        "operationId": "getTestByName",
        "parameters": [
          {
            "in": "path",
            "name": "name",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "properties": {
                    "cases": {
                      "items": {
                        "properties": {
                          "command": {
                            "type": "string"
                          },
                          "name": {
                            "type": "string"
                          }
                        }
                      },
                      "type": "array"
                    },
                    "name": {
                      "type": "string"
                    },
                    "verifier": {
                      "properties": {
                        "name": {
                          "type": "string"
                        }
                      }
                    }
                  }
                }
              }
            },
            "description": "Success"
          }
        },
        "summary": "Get test",
        "tags": [
          "test"
        ]
      }
    }
  },
  "servers": [
    {
      "url": "https://test.oapi-codegen.com/v2"
    },
    {
      "url": "http://test.oapi-codegen.com/v2"
    }
  ]
}
